draft-birkholz-cose-cometre-ccf-profile.md

```diff
@@ -98,29 +111,42 @@ where:
 - : denotes concatenation of lists
 - D\[k1:k2\] = D'_(k2-k1) denotes the list \{d'\[0\] = d\[k1\], d'\[1\] = d\[k1+1\], ..., d'\[k2-k1-1\] = d\[k2-1\]\} of length (k2 - k1).
 
-## Leaf Components
+## Transaction Components
 
 Each leaf in a CCF ledger carries the following components:
 
 ~~~
 CCF-leaf = [
-  internal-hash: bstr ; a string of HASH_SIZE bytes;
-  internal-data: bstr; a string of at most 1024 bytes; and
-  data_hash: bstr ; the serialization of the element stored at this leaf.
+
+  ; a string of HASH_SIZE bytes
+  internal-transaction-hash: bstr
+
+  ; a string of at most 1024 bytes;
+  internal-evidence: bstr
+
+  ; a string of HASH_SIZE bytes
+  data-hash: bstr
 ]
 ~~~
 
-The `internal_hash` and `internal_data` byte strings are internal to the CCF implementation. Similarly, the auxiliary tree entries are internal to CCF. They are opaque to receipt Verifiers, but they commit the TS to the whole tree contents and may be used for additional, CCF-specific auditing.
+The `internal-transaction-hash` and `internal-evidence` byte strings are internal to the CCF implementation. They can be safely ignored by receipt Verifiers, but they commit the TS to the whole tree contents and may be used for additional, CCF-specific auditing.
 
 # CCF Inclusion Proofs
 
 CCF inclusion proofs consist of a list of digests tagged with a single left-or-right bit.
 
 ~~~
-CCF-inclusion-proof: [+ proof-element],
+CCF-inclusion-proof: [
```

Comment:
labels/keys in arrays are intended to improve readability for humans and are not conveyed in the end. Was the intend maybe to use a map here, instead?

If so, is that also a possibility for other arrays that include elements with string label definitions (e.g., CCF-leaf that looks to me more like an annotation for humans that read the CDDL definition, but now I am not so sure)?

As all elements are mandatory, an array is an okay choice in any case. Just unsure about the intend of the labels.

Reply:
I think I prefer an array over a map here, repeating the keys seems quite wasteful considering this is not a self-describing document. I think I should have used comments here rather than labels, let me fix that.

Reply:
Sorry read that too quickly, a map does seem preferable at the top-level.